lib/Logger.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * @package ActiveRecord
+ */
+namespace ActiveRecord;
+
+/**
+ * Defines a logger for logging queries.
+ *
+ * @package ActiveRecord
+ */
+interface Logger
+{
+
+	/**
+	 * Logs a message.
+	 *
+	 * Implementations should expect that `done()` will be called after a call
+	 * to this method, for logged SQL queries.
+	 *
+	 * @param string $message The message to log.
+	 * @param array $params Any parameters associated with the message.
+	 * @return void
+	 */
+	public function log($message, array $params = null);
```

Comment:
Yea, I purposefully _didn't_ copy the Doctrine API here. I had originally done so, but in order to follow the style of the AR project, the method names needed to be underscored-separated (`start_query()`), which would break style for implementations created in PSR-2 compliant projects. So I purposefully named these methods with single word names. I briefly mentioned this in our chat the other day. Make sense?